File: Submissions/822_Card_Flipping_Game.cpp
class Solution {
public:
    int flipgame(vector<int>& fronts, vector<int>& backs) {
        int minimum = 2001;
        set<int> s;
        for(int i=0; i<fronts.size(); ++i){
            if(fronts[i] == backs[i]) s.insert(fronts[i]);
        }
        for(int i=0; i<fronts.size(); ++i){
            if(s.find(fronts[i])==s.end()) minimum = min(minimum, fronts[i]);
            if(s.find(backs[i])==s.end()) minimum = min(minimum, backs[i]);
        }
        return (minimum==2001) ? 0 : minimum;
    }
};

File: Kotlin/822_Card_Flipping_Game.kt
class Solution {
    fun flipgame(fronts: IntArray, backs: IntArray): Int {
        var minimum = 2001
        val s = mutableSetOf<Int>()
        for(i in 0 until fronts.size){
            if(fronts[i] == backs[i]) s.add(fronts[i])
        }
        for(i in 0 until fronts.size){
            if(!s.contains(fronts[i])) minimum = minOf(minimum, fronts[i])
            if(!s.contains(backs[i])) minimum = minOf(minimum, backs[i])
        }
        
        return if(minimum == 2001) 0 else minimum
    }
}